### ISSUE TITLE
Chaining/inheritance/sourcing?

### ISSUE THREAD

**Issue by v4eddie** [theskumar/python-dotenv#37]
I think it would be handy to provide a way for a `.env` file to include a "base" file with a bunch of default values, that it can then override. A couple thoughts on how it might work:

* Use "source", like in bash. Just put this line at the top of `.env`

`
source "./defaults.env"
`

And all the values in `defaults.env` get set as if they were entered in `.env` itself, with later entries overriding the earlier ones. You would probably also want an `unset` keyword to remove inherited values.

That's probably the simplest approach. If you wanted to get fancy, you could make the inheritance explicit. Something like:

```
base "./defaults.env"

MY_VALUE="xyz"          # Sets this variable, as normal
inherit OTHER_VALUE   # Sets this variable to the value specified in defaults.env. Doesn't set any other values.
```

In other words, you would have to explicitly list which variables to inherit. You could also have `inherit all` and `unset` keywords.

I would probably lean towards the first approach, just for simplicity's sake.
Thoughts? I'd be happy to contribute a patch.

**Comment by hugochinchilla** [comment]
You can handle this in your application code by loading a `.defaults.env` first and then a `.env`

**Comment by theskumar** [comment]
Like @hugochinchilla mention, this is quite trivial thing to do and not necessarily need any modification in the codebase do to it. 

@v4eddie I'm afraid we would be able able to include this feature, unless we have substantial people requiring it. 

Cheers,

**Comment by theskumar** [comment]
I'm closing this ticket, since i don't see anyone interested after 19days of waiting.

**Comment by twosigmajab** [comment]
Just found this issue. FWIW I think this would be very useful. It's a violation of 12-factor principles to let the logic of deciding how to populate the environment bleed into the application layer. The application layer should only read from the environment. An outer layer like `dotenv run` is where this logic should go, so I think it's reasonable to consider more thoughtfully whether dotenv should provide this feature. One workaround in the meantime that relies on a bash feature is something like `dotenv -f <(cat base.env overrides.env) run ...`.
